tracer/src/Datadog.Trace/DataStreamsMonitoring/SpanExtensions.cs

```diff
@@ -0,0 +1,25 @@
+// <copyright file="SpanExtensions.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+namespace Datadog.Trace.DataStreamsMonitoring;
+
+internal static class SpanExtensions
+{
+    /// <summary>
+    /// Sets a DataStreams checkpoint and adds pathway tag to the span
+    /// </summary>
+    /// <param name="span">The span instance</param>
+    /// <param name="manager">The <see cref="DataStreamsManager"/> to use</param>
+    /// <param name="edgeTags">The edge tags for this checkpoint. NOTE: These MUST be sorted alphabetically</param>
+    internal static void SetDataStreamsCheckpoint(this Span span, DataStreamsManager manager, string[] edgeTags)
+    {
+       span.Context.SetCheckpoint(manager, edgeTags);
+       var hash = span.Context.PathwayContext?.Hash.Value ?? 0;
+       if (hash != 0)
+       {
+           span.SetTag("pathway.hash", hash.ToString());
```

Comment:
NOTE: if we enable DSM monitoring by default at some point, we should probably add a `PathwayHash` property to the `KafkaTags` type, so we can "fast path" this instead of using the slower `SetTag()` API. 

I don't think it's worth doing yet, given DSM is optional, as it would increase allocations very slightly when DSM is disabled, but it's a possibility. e.g. If we added it, we could do something like this:

```csharp
           if(span.Tags is KafkaTags kafkaTags)
           {
               kafkaTags.PathwayHash = hash.ToString();
           }
           else
           {
               span.SetTag("pathway.hash", hash.ToString());
           }
```